src/Raven.Server/Utils/PoolOfThreads.cs

```diff
@@ -123,10 +133,81 @@ public void LowMemoryOver()
             _lowMemoryFlag.Lower();
         }
 
+        private static void ReleaseThread(PooledThread pooled)
+        {
+            pooled.InPoolSince = null;
+            pooled.SetWorkForThread(null, null, null, null);
+        }
+
         private void Clear()
         {
-            while (_pool.TryDequeue(out var pooled))
-                pooled.SetWorkForThread(null, null, null, null);
+            while (_pool.TryPop(out var pooled))
+                ReleaseThread(pooled);
+        }
+
+        private void Cleanup(object _)
+        {
+            const int minNumberOfItemsInPoolToPerformCleanupCheck = 64;
```

Comment:
was thinking about this, but we do:
```
var thread = new Thread(pooled.Run, PlatformDetails.Is32Bits ? 512 * Constants.Size.Kilobyte : 0)
```

which means that threads already allocate less there, so the memory footprint value is already much lower than on 64bits, that is why I didn't want to cut this by half, because on x86 this would be actual 1/4 memory footprint of x64